composer/core/data_spec.py

```diff
@@ -155,11 +155,20 @@ class DataSpec:
         num_tokens (int, optional): The total number of tokens in an epoch. This field is used by the
             :class:`.Timestamp` (training progress tracker).
 
-        device_transforms ((Batch) -> Batch, optional): Function called by the :class:`.Trainer` to modify the
-            batch once it has been moved onto the device. For example, this function can be used for GPU-based
+        device_transforms ((Batch) -> Batch, optional): Deprecated argument. Please use ``batch_transforms`` for batch
+            level transformations on CPU and ``microbatch_transforms`` for microbatch level transformations on target
+            device.
```

Comment:
Should `device_transforms` be kept but applied to the micro-batches. The name remains valid because the transformation is performed after the micro-batch is moved to the requested device. This would eliminate the need for deprecating this argument.

Reply:
I'm for renaming it to make it more clear where it runs. I'm concerned about subtle logic bugs that may emerge otherwise as people do not realize it does not run at batch level.